pkg/apis/serving/v1alpha2/inferenceservice_validation.go

```diff
@@ -18,18 +18,25 @@ package v1alpha2
 
 import (
 	"fmt"
+
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // Known error messages
 const (
-	MinReplicasShouldBeLessThanMaxError = "MinReplicas cannot be greater than MaxReplicas."
-	MinReplicasLowerBoundExceededError  = "MinReplicas cannot be less than 0."
-	MaxReplicasLowerBoundExceededError  = "MaxReplicas cannot be less than 0."
-	TrafficBoundsExceededError          = "TrafficPercent must be between [0, 100]."
-	TrafficProvidedWithoutCanaryError   = "Canary must be specified when CanaryTrafficPercent > 0."
-	UnsupportedStorageURIFormatError    = "storageUri, must be one of: [%s] or match https://{}.blob.core.windows.net/{}/{} or be an absolute or relative local path. StorageUri [%s] is not supported."
+	MinReplicasShouldBeLessThanMaxError     = "MinReplicas cannot be greater than MaxReplicas."
+	MinReplicasLowerBoundExceededError      = "MinReplicas cannot be less than 0."
+	MaxReplicasLowerBoundExceededError      = "MaxReplicas cannot be less than 0."
+	TrafficBoundsExceededError              = "TrafficPercent must be between [0, 100]."
+	TrafficProvidedWithoutCanaryError       = "Canary must be specified when CanaryTrafficPercent > 0."
+	UnsupportedStorageURIFormatError        = "storageUri, must be one of: [%s] or match https://{}.blob.core.windows.net/{}/{} or be an absolute or relative local path. StorageUri [%s] is not supported."
+	CPURequestShouldBeLessThanLimitError    = "CPU request cannot be greater than CPU limit."
```

Comment:
I would generalize all of these into RequestsBoundsExceededeError "Request must be greater than 0 and less than the limit."

Reply:
Now I introduced the k8s.io/kubernetes and used the k8s.io/kubernetes/pkg/apis/core/validation/validation.go:ValidateResourceRequirements so all the message are from the api itself.